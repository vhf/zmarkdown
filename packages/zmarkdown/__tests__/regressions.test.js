const a = require('axios')
const dedent = require('dedent')

const u = (path) => `http://localhost:27272${path}`
const html = u('/html')
const latex = u('/latex')

describe('Regression tests', () => {
  describe('HTML endpoint', () => {
    test('#188 It does not crash on unsupported languages fenced code blocks', async () => {
      const response = await a.post(html, {
        md: [
          '# foo',
          '',
          '```foobar',
          'console.error("foo", true)',
          '```',
          '',
          '```js',
          'console.error("foo", true)',
          '```',
        ].join('\n'),
        opts: {},
      })

      const [rendered] = response.data
      expect(rendered).toMatchSnapshot()
    })

    test.skip('SVG camelCase attributes', async () => {
      // skipped because rehype-katex doesn't need an SVG fracline for this anymore
      const response = await a.post(html, {
        md: [
          '$\\hat \\theta_{MC} = \\frac{\\sum_{i = 1}^n P(Y_i | X = 1)}{\\pi^3}$',
          '',
          '### normal',
          '',
          '$$\\hat \\theta_{MC} = \\sqrt{\\frac{\\sum_{i = 1}^n P(Y_i | X = 1)}{\\pi^3}}$$',
          '',
          '### display',
          '',
          '$$',
          '\\hat \\theta_{MC} = \\frac{\\sum_{i = 1}^n P(Y_i | X = 1)}{\\pi^3}',
          '$$',
        ].join('\n'),
        opts: {},
      })

      const [rendered] = response.data
      expect(rendered).toContain('viewBox')
      expect(rendered).not.toContain('preserve-aspect-ratio')
      expect(rendered).toContain('preserveAspectRatio')
      expect(rendered).not.toContain('preserve-aspect-ratio')
    })
  })

  describe('Latex endpoint', () => {
    test('It wraps image basenames containing dots', async () => {
      const response = await a.post(latex, {
        md: dedent`
          ![](x.yz.png)

          [![foo](/a/w.x.y.z.png)](http://example.com)

          ![](/w.x.y.z.png)

          ![](/foo.bar/x.yz.png)
        `,
        opts: {},
      })

      const [rendered] = response.data
      expect(rendered).toMatchSnapshot()
    })
  })
})
