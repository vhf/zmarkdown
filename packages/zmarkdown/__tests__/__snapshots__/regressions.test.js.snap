// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regression tests HTML endpoint #188 It does not crash on unsupported languages fenced code blocks 1`] = `
"<h3 id=\\"foo\\">foo<a aria-hidden=\\"true\\" href=\\"#foo\\"><span class=\\"icon icon-link\\"></span></a></h3>
<div class=\\"hljs-code-div\\"><div class=\\"hljs-line-numbers\\"><span></span></div><pre><code class=\\"hljs language-foobar\\">console.error(\\"foo\\", true)
</code></pre></div>
<div class=\\"hljs-code-div\\"><div class=\\"hljs-line-numbers\\"><span></span></div><pre><code class=\\"hljs language-js\\"><span class=\\"hljs-built_in\\">console</span>.error(<span class=\\"hljs-string\\">\\"foo\\"</span>, <span class=\\"hljs-literal\\">true</span>)
</code></pre></div>"
`;
